pkg/providers/tinkerbell/tinkerbell.go

```diff
@@ -71,7 +73,11 @@ type Provider struct {
 	Retrier          *retrier.Retrier
 }
 
-// TODO: Add necessary kubectl functions here
+type ProviderDockerClient interface {
```

Comment:
I think this could be better named. Perhaps just `Docker` or `DockerRuntime`?

Reply:
I just followed how other executables are named like `ProviderKubectlClient`. I'm fine with just changing it to `Docker` but we should also change the others as well for consistency.